doc/source/whatsnew/v0.24.0.txt

```diff
@@ -194,6 +194,7 @@ Other Enhancements
 - :meth:`Index.to_frame` now supports overriding column name(s) (:issue:`22580`).
 - New attribute :attr:`__git_version__` will return git commit sha of current build (:issue:`21295`).
 - Compatibility with Matplotlib 3.0 (:issue:`22790`).
+- Added :meth:`Interval.overlaps`, :meth:`IntervalArray.overlaps`, and :meth:`IntervalIndex.overlaps` for determining overlaps between interval-like objects (:issue:`21998`)
```

Comment:
I don't think IntervalArray is in our API docs, so that link will cause a warning.

Reply:
It is not in the API docs at this time, but I imagine we'd add it if we decide to make `IntervalArray` (and other EA's) public?  I have a few doc related TODOs surrounding whether or not `IntervalArray` gets made public, so I'll make an issue for these TODOs and include the above in the next day or so.